client/driver/fabfile.py

```diff
@@ -150,8 +181,18 @@ def loop():
     # run oltpbench as a background job
     run_oltpbench_bg()
 
-    # run controller
-    run_controller()
+    # run controller from another process
+    p = Process(target=run_controller, args=())
+    while not _ready_to_start_controller():
+        pass
+    p.start()
+    while not _ready_to_shut_down_controller():
+        pass
+    # stop the experiment
+    stop_controller()
+
+    # to prevent the race condition, wait for the controller to wrap up
+    time.sleep(20)
```

Comment:
Is there a better way to do this? something like p.wait() in Python's multiprocess.

Reply:
Please look into  p.join()